DevAcademy/Services/DataService.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  DataService.swift
+//  DevAcademy
+//
+//  Created by Tomáš Melichar on 18.07.2023.
+//
+
+import Foundation
+
+private enum ServiceError: Error {
+    case noData
+    case fatalError
+}
+
+final class DataService {
+    static let shared = DataService()
+    
+    var data: Result<Features, Error>?
+    
+    private init() {}
+    
+    func fetchData(closure: @escaping (Result<Features, Error>) -> ()) {
+        if let data = self.data {
+            closure(data)
+            return
+        } else {
+            Timer.scheduledTimer(withTimeInterval: 2.2, repeats: false) { [weak self] _ in
+                self?.data = .success(DataService.mockData)
```

Comment:
posledni drobnost, po nastaveni tech dat je potreba ty data i vratit v predane `closure` (aby se s kazdym zavolanim teto funkce zavolala i `closure`).

Takze asi nejak takhle:
```Swift
Timer.scheduledTimer(withTimeInterval: 2.2, repeats: false) { [weak self] _ in
       let result = .success(DataService.mockData)
       self?.data = result
       closure(result)
}
```

Pouzivam pomocnou promennou `result`, abych ty nastavena `self?.data` nemusel unwrappovat, protoze jsou v DataService jako optional. 😉

Reply:
Upravil jsem to, Jen jsem musel pozměnit `let result: Result<Features, Error> = .success(DataService.mockData)` - kvůli "Reference to member 'success' cannot be resolved without a contextual type" 😳